crates/bitcoin/src/types.rs

```diff
@@ -263,22 +263,25 @@ impl BlockHeader {
     }
 }
 
+#[derive(PartialEq, Clone, Debug)]
+pub enum TransactionInputSource {
```

Comment:
What would it look like if we separated `TransactionInputCoinbase` and kept `TransactionInput` mostly intact?

Reply:
Well, different.  You'd have to extract the common fields into a struct if you don't want duplication. I think the current change is somewhat cleaner